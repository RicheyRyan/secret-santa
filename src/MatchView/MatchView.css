.match-list{
  width: 100%;
  box-sizing: border-box;
  padding: 0 2em;
  list-style: none;
}

.match-list li {
  width: 100%;
  display: block;
  padding-bottom: 1em;
}

.match-name, .match-middle-text{
  display: inline-block;
  text-align: center;
  width: 33%;
}

.match-name {
  font-size: 1.4em;
}

.match-middle-text{
  font-weight: bold;
  font-size: .8em;
  margin-top: .5em;
}
