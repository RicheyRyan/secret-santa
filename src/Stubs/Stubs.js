export const users = {
  items: [
    {
      guid: '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
      name: {
        first: 'Jennifer',
        last: 'Madden',
      },
      email: 'jmadser@somemail.com',
      phone: '+353823452345',
    },
    {
      guid: 'f45e4343-6dfb-4085-b249-975e8d707b2e',
      name: {
        first: 'Bob',
        last: 'Bertler',
      },
      email: 'bb6699@nowhere.com',
      phone: '+353880456445',
    },
    {
      guid: '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
      name: {
        first: 'Lisa',
        last: 'Masterson',
      },
      email: 'lisam76@hotmail.com',
      phone: '+353823423423',
    },
    {
      guid: '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
      name: {
        first: 'Nelson',
        last: '',
      },
      email: 'nelsonator16@themail.com',
      phone: '+353856456456',
    },
    {
      guid: '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
      name: {
        first: 'Fred',
        last: 'Walsh',
      },
      email: 'fred.walsh82@gmail.com',
      phone: '+353878234567',
    },
    {
      guid: '62d2c77e-6297-43b7-864c-042ca1ee715d',
      name: {
        first: 'Geoff',
        last: 'Lanegan',
      },
      email: 'glanegan@laneganers.ie',
      phone: '+353845645645',
    },
    {
      guid: 'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
      name: {
        first: 'Brian',
        last: 'Sullivan',
      },
      email: 'bsull@hmail.co.uk',
      phone: '+353823456776',
    },
    {
      guid: '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
      name: {
        first: 'Lisa',
        last: 'Desmond',
      },
      email: 'ldesmond1@shhmail.com',
      phone: '+353834543543',
    },
    {
      guid: 'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
      name: {
        first: 'Derek',
        last: 'Proctor',
      },
      email: 'theproctor@zmail.ie',
      phone: '+353878634234',
    },
  ],
  byId: {
    '7302a5c9-f12e-45e7-a376-a9f05e4b88fc': {
      guid: '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
      name: {
        first: 'Jennifer',
        last: 'Madden',
      },
      email: 'jmadser@somemail.com',
      phone: '+353823452345',
    },
    'f45e4343-6dfb-4085-b249-975e8d707b2e': {
      guid: 'f45e4343-6dfb-4085-b249-975e8d707b2e',
      name: {
        first: 'Bob',
        last: 'Bertler',
      },
      email: 'bb6699@nowhere.com',
      phone: '+353880456445',
    },
    '39bb60c9-cc52-4edd-a213-a8176f2d4d1c': {
      guid: '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
      name: {
        first: 'Lisa',
        last: 'Masterson',
      },
      email: 'lisam76@hotmail.com',
      phone: '+353823423423',
    },
    '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35': {
      guid: '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
      name: {
        first: 'Nelson',
        last: '',
      },
      email: 'nelsonator16@themail.com',
      phone: '+353856456456',
    },
    '51944eee-7eb4-4ecb-99e2-981d05f85e1e': {
      guid: '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
      name: {
        first: 'Fred',
        last: 'Walsh',
      },
      email: 'fred.walsh82@gmail.com',
      phone: '+353878234567',
    },
    '62d2c77e-6297-43b7-864c-042ca1ee715d': {
      guid: '62d2c77e-6297-43b7-864c-042ca1ee715d',
      name: {
        first: 'Geoff',
        last: 'Lanegan',
      },
      email: 'glanegan@laneganers.ie',
      phone: '+353845645645',
    },
    'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b': {
      guid: 'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
      name: {
        first: 'Brian',
        last: 'Sullivan',
      },
      email: 'bsull@hmail.co.uk',
      phone: '+353823456776',
    },
    '587bf323-5b7b-43ca-a88f-a1e9c884c7cc': {
      guid: '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
      name: {
        first: 'Lisa',
        last: 'Desmond',
      },
      email: 'ldesmond1@shhmail.com',
      phone: '+353834543543',
    },
    'cd3515f2-9b52-42ea-9f3c-e6daaf65249a': {
      guid: 'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
      name: {
        first: 'Derek',
        last: 'Proctor',
      },
      email: 'theproctor@zmail.ie',
      phone: '+353878634234',
    },
  },
};
export const matches = {
  '587bf323-5b7b-43ca-a88f-a1e9c884c7cc':
    '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35',
  '2aaa1a3d-d56c-45a7-b1f5-c6d57626ba35':
    '7302a5c9-f12e-45e7-a376-a9f05e4b88fc',
  '7302a5c9-f12e-45e7-a376-a9f05e4b88fc':
    '62d2c77e-6297-43b7-864c-042ca1ee715d',
  '62d2c77e-6297-43b7-864c-042ca1ee715d':
    'f45e4343-6dfb-4085-b249-975e8d707b2e',
  'f45e4343-6dfb-4085-b249-975e8d707b2e':
    'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b',
  'f26c5535-e9fa-4a44-aee7-d0e5d9219e8b':
    '51944eee-7eb4-4ecb-99e2-981d05f85e1e',
  '51944eee-7eb4-4ecb-99e2-981d05f85e1e':
    'cd3515f2-9b52-42ea-9f3c-e6daaf65249a',
  'cd3515f2-9b52-42ea-9f3c-e6daaf65249a':
    '39bb60c9-cc52-4edd-a213-a8176f2d4d1c',
  '39bb60c9-cc52-4edd-a213-a8176f2d4d1c':
    '587bf323-5b7b-43ca-a88f-a1e9c884c7cc',
};
