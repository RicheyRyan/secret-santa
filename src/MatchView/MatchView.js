import React from 'react';
import './MatchView.css';
import Loading from '../Loading/Loading.js';
import Cta from '../Cta/Cta.js';
import { getUsersName } from '../Utils/user.js';

const MatchView = props =>
  props.matches
    ? <div>
        <ul className="match-list">
          {props.users.items.map(user => {
            const { guid } = user;
            const matchId = props.matches[guid];
            const matchUser = props.users.byId[matchId];
            return (
              <li key={guid}>
                <div className="match-name">{getUsersName(user)}</div>
                <div className="match-middle-text">was matched with</div>
                <div className="match-name">{getUsersName(matchUser)}</div>
              </li>
            );
          })}
        </ul>
        <Cta
          label={'Reshuffle'}
          message={'Dont like the matches?'}
          onClick={props.reshuffle}
        />
        <Cta
          label={'Preview Emails'}
          message={'Happy?'}
          onClick={props.toggleEmailView}
        />
      </div>
    : <Loading />;

export default MatchView;
