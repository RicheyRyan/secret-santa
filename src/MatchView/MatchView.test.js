import React from 'react';
import renderer from 'react-test-renderer';
import MatchView from './MatchView.js';
import { users, matches } from '../Stubs/Stubs.js';

it('Cta component matches snapshot', () => {
  const component = renderer.create(
    <MatchView
      users={users}
      matches={matches}
      reshuffle={() => {}}
      toggleEmailView={() => {}}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
