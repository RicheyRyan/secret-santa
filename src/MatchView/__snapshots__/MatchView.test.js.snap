exports[`test Cta component matches snapshot 1`] = `
<div>
  <ul
    className="match-list">
    <li>
      <div
        className="match-name">
        Jennifer Madden
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Geoff Lanegan
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Bob Bertler
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Brian Sullivan
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Lisa Masterson
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Lisa Desmond
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Nelson 
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Jennifer Madden
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Fred Walsh
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Derek Proctor
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Geoff Lanegan
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Bob Bertler
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Brian Sullivan
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Fred Walsh
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Lisa Desmond
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Nelson 
      </div>
    </li>
    <li>
      <div
        className="match-name">
        Derek Proctor
      </div>
      <div
        className="match-middle-text">
        was matched with
      </div>
      <div
        className="match-name">
        Lisa Masterson
      </div>
    </li>
  </ul>
  <div
    className="cta">
    <p>
      Dont like the matches?
    </p>
    <button
      onClick={[Function]}>
      Reshuffle
    </button>
  </div>
  <div
    className="cta">
    <p>
      Happy?
    </p>
    <button
      onClick={[Function]}>
      Preview Emails
    </button>
  </div>
</div>
`;
